mantidproject/mantid
# Engineering Diffraction GUI can load and plot multiple runs

Loading and plotting multiple focused runs in the fitting tab of the ED GUI has been implemented. In order to achieve this, work was started on implementing an explicit model (as opposed to essentially using the ADS) for the fitting tab.

Note: the fitting tab is a work in progress. Attempting to do a fit will probably crash Mantid, and some of the code is a little rough around the edges as I haven't yet moved all the model-ly stuff into the model

**To test:**
[Re-enable the GUI](https://github.com/mantidproject/mantid/commit/8011de68c96312cf2ccf746ac34c8d433a07534e) and build.

Some sample data is [here](https://github.com/mantidproject/mantid/files/1460806/ENGINX_focused_runs.zip).

Open the ED GUI (Interfaces > Diffraction > Engineering Diffraction) and go to the fitting tab. Click the first Browse button and select all (or some combination of) the attached files, then hit Load. The loaded workspaces should appear in the list next to the plot, labelled with run number and bank number. Clicking on these should plot the corresponding workspace. Try and break things, and also try renaming the files for run 277208 and seeing if the load still works.

Partially fixes #21082 

**Release Notes** 

The change isn't complete, so shouldn't go in the release notes. I'll add it once fitting is wired up

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
